gpkit/constraints/bounded.py

```diff
@@ -0,0 +1,68 @@
+"Implements BoundedConstraintSet"
+from collections import defaultdict
+import numpy as np
+
+from .. import Variable
+from .set import ConstraintSet
+from ..small_scripts import mag
+
+
+def add_bounds(varkeys, lower, upper):
+    "Returns constraints list bounding all varkeys"
+    constraints = []
+    for varkey in varkeys:
+        units = varkey.descr.get("units", 1)
+        constraints.append([upper*units >= Variable(**varkey.descr),
+                            Variable(**varkey.descr) >= lower*units])
+    return constraints
+
+
+class BoundedConstraintSet(ConstraintSet):
+    "Bounds contained variables so as to ensure dual feasibility."
+
+    def __init__(self, constraints, substitutions=None, verbosity=1,
+                 eps=1e-30, lower=None, upper=None):
+        self.bound_las = None
+        self.verbosity = verbosity
+        self.lowerbound = lower if lower else eps
+        self.upperbound = upper if upper else 1/eps
+        constraintset = ConstraintSet([constraints])
```

Comment:
what if `constraints` is a `TightConstraintSet` or other child of `ConstraintSet`? Won't this demote it to a `ConstraintSet`?

Another potential way: should `ConstraintSet` have an `add_constraints` method, so this could call super init without casting `constraints` to `ConstraintSet`, and then add the bounding constraints?

Alternatively, if the cast is just a handy way to get the varkeys, then I think the line below 
```constraints = [bounding_constraints, constraintset]```
should use `constraints` in place of `constraintset`.